store/state/integer.go

```
@@ -0,0 +1,48 @@
+package state
+
+type Integer struct {
+	Value
+
+	enc IntEncoding
+}
+
+func NewInteger(v Value, enc IntEncoding) Integer {
+	return Integer{
+		Value: v,
+		enc:   enc,
+	}
+}
+
+func (v Integer) Get(ctx Context) (res uint64) {
+	bz := v.GetRaw(ctx)
+	if bz == nil {
+		return 0
+	}
+	res, err := DecodeInt(bz, v.enc)
+	if err != nil {
+		panic(err)
+	}
+	return res
+}
+
+func (v Integer) GetSafe(ctx Context) (uint64, error) {
+	bz := v.GetRaw(ctx)
+	if bz == nil {
+		return 0, &GetSafeError{}
+	}
+	res, err := DecodeInt(bz, v.enc)
+	if err != nil {
+		panic(err)
```

Why are we panic'ing here? We should rather return an error, no?